daemon/cmd/endpoint.go

```diff
@@ -282,7 +282,7 @@ func (m *endpointCreationManager) DebugStatus() (output string) {
 
 // createEndpoint attempts to create the endpoint corresponding to the change
 // request that was specified.
-func (d *Daemon) createEndpoint(ctx context.Context, epTemplate *models.EndpointChangeRequest) (*endpoint.Endpoint, int, error) {
+func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, epTemplate *models.EndpointChangeRequest) (*endpoint.Endpoint, int, error) {
```

Comment:
I don't understand this change, why can't we keep the old way and still pass `d` to the functions on line 316, 412 and 497?

Reply:
This change allows us to pass `DaemonSuite` as the `owner`, because it implements the `regeneration.Owner` interface. This is useful because we can mock out certain calls that happen during the creation of the endpoint. Specifically, to fix the metrics assertion flake (as mentioned in the commit msg and in the PR body), we want to mock out the call to `(*Daemon).QueueEndpointBuild` because that causes the endpoint to transition states extremely quickly. See [this comment](https://github.com/cilium/cilium/issues/11955#issuecomment-646340820) for more info on this.

Does that make sense? Happy to clarify further.

Reply:
oh I miss that we were passing `ds` in the tests and not `ds.d`